### ISSUE TITLE
Should pages be denoted as editable by the community?

### ISSUE THREAD

**Issue by MTres19** [MTres19/stk-docs#5]
I was thinking about how the documentation could be maintained. Currently, it's not obvious that the website runs on MediaWiki, which might discourage potential contributors. I was thinking that each wiki page (except the homepage and other pages that aren't documentation) could have a banner (using MediaWiki templates) at the top placing it in one of the following catagories:

* **Community:** The community is free to contribute by editing and improving these pages, as long as their edits do not contradict guidelines. (Examples: Making Tracks, Making Karts, Making Textures, Scripting)
* **Restricted:** The community may contribute to these pages, but must first get approval from an STK admin. (Examples: Getting To Know the Style, Building on Windows/Linux/macOS, Music & SFX Specifications)
* **Locked:** Only admins may modify these pages, and the community should not contact the team with suggestions unless a serious flaw has been found. (Examples: Source Control, Licensing, To Do, Needed Objects, Coding Style Guidelines)

What's your opinion? For this, of course, allowing users to sign up would have to be enabled, but I think the [No Captcha Recaptcha](https://www.google.com/recaptcha/intro/index.html) is pretty good for spam prevention.

**Comment by samuncle** [comment]
Actually it was how the old website was managed and a few things happened:

1) a lot of pages where Unmaintened which is a problem because we must provide accurate information to the user

2) some pages where irrelevant, things that are more suitable for a fan website 

3) everything looked a bit unorganised and anarchist to the newcomers

---

We decided to try to make the website a bit more restricted and to only allow users who are registered and willing to write articles.
Now maybe we could show to people they can suggest modifications. Actually I really like your idea of a github for documentation.

People can suggest things and after being reviewed they are included in the documentation.

I prefer to avoid being a clone of Wikipedia (design and philosophy) because we aren't exactly a wiki. It's our official website and wiki look all the same. 

Now I think we could add something like:

Have you a suggestion? You want to submit an article? 
Read more here (with link to a github)

**Comment by MTres19** [comment]
Yeah, that probably is a better idea. With a quick Google search, I found [this](https://github.com/moy/Git-Mediawiki/wiki/User-manual). It looks like a cron job or systemd unit could be set up to automatically git pull --rebase from GitHub and then push to the wiki. Then changes could be sent in via pull requests and approved or denied before they get into the wiki.

**Comment by qwertychouskie** [comment]
I love the new site, but also have noticed many small thing that need to be fixed/tweaked, e.g. grammar issues. I would be happy to contribute fixes, but for that to work there would need to be some sort of system for contributions.  It would be nice to get a system soon, so I can start contributing. :)

**Comment by MTres19** [comment]
@qwertychouskie Grammar issues are probably the least of the developers' worries. Not that they shouldn't be fixed if you or I get access to the wiki, but if you want to help, learn MediaWiki markup (it's not hard) and help me write documentation, which is really lacking right now. The Communication article (Communication.txt in this repo) is still empty. (See  issue #3.) Write a draft and send it in via a pull request, if you're interested in contributing.

Regards,
-

**Comment by konstin** [comment]
Using git-mediawiki is definitely an interesting idea. You can try syncing the wiki with github through your local machine to a github repository to see if or how good this works. In case it works flawless, we could use this in production, though I won't make any promises in that regard.

**Comment by MTres19** [comment]
Yes, I'll give it a try when I have a chance.

**Comment by MTres19** [comment]
Okay, I'm able to clone, and I think it could be set up so that it would be possible to fetch from GitHub and push to MediaWiki. What might be more difficult is setting up a system so that things could be edited on either GitHub or MediaWiki, and resolving merge conflicts and such. I'm actually getting less convinced that this is a good idea.

**Comment by konstin** [comment]
> What might be more difficult is setting up a system so that things could be edited on either GitHub or MediaWiki, and resolving merge conflicts and such.

Just define one of the two as superior, so in case there's a conflict this version will be taken. I'ld say github wins, but that's open for discussion.

> I'm actually getting less convinced that this is a good idea.

It is and will always be a workaround, but it could make things much easier.

**Comment by MTres19** [comment]
I finally managed to come up with a script that should work somewhat:

<pre>#!/bin/bash
#
# Auto sync script for the SuperTuxKart wiki

# Set up repo if not already done
if ! [ -d "supertuxkart.net" ]
then
    echo "Setting up repository..."
    
    git clone --origin wiki mediawiki::https://supertuxkart.net
    cd "supertuxkart.net"
    git remote add github https://github.com/MTres19/supertuxkart.net.git
    git push github master
fi


cd "supertuxkart.net"

git pull --rebase wiki
git pull --rebase -s recursive -X ours github master

git push wiki master
git push github master</pre>

I also created the user "GitHub-Sync" on the wiki which I plan to use for pushing to the wiki. However, three things are still needed:

* A bot user for GitHub so I'm not using my own credentials in a script
* A way to include the usernames and passwords of the bot users in the script
* A way to run this on the VPS, because I don't really have the infrastructure to run this myself.

**Comment by konstin** [comment]
The script looks fine so far, there will only need to be some minor environment specific changes. Creating an additional user won't be necessary as github allows using special tokens with limited rights. The token can than be passed to the script as a command line argument.

When if tested everything I'll set it up on the VPS. Unfortunatly this has to wait for a week as I'm currently on holidays and therefore can't access the VPS.

btw the tool used for deployment is https://github.com/logsol/Github-Auto-Deploy , which runs as a systemd service. It is pretty cool as it's just one small python script that you can configure to do anything.

**Comment by MTres19** [comment]
Okay. Thanks.

**Comment by MTres19** [comment]
@konstin Do you need the password for the GitHub-Sync user?

**Comment by MTres19** [comment]
I've started work on a (page)[https://supertuxkart.net/Maintaining_the_Wiki] about contributing, but it's not yet done.

**Comment by MTres19** [comment]
@konstin I edited the script because things need to be pulled from the wiki for the wiki to allow the script to push, and things need to be pulled from GitHub for GitHub to allow the script to push. In a way, they're mutually partially incompatible.

I kept the first two git pulls because I want merge conflicts resolved in the current wiki revision's favor, not an old revision.

**Comment by MTres19** [comment]
You'll need to add the following properties to .git/config, too:
<pre>
mwLogin = GitHub-Sync
mwPassword = **********
mediaimport = true
</pre>
No, that's not the actual password for that user.

**Comment by MTres19** [comment]
@konstin I updated the script and my comment above (again). Just to automatically import files.
